pkg/manager/impl/dataset_manager.go

```diff
@@ -129,6 +132,58 @@ func (dm *datasetManager) GetDataset(ctx context.Context, request datacatalog.Ge
 	}, nil
 }
 
+// List Datasets with optional filtering and pagination
+func (dm *datasetManager) ListDatasets(ctx context.Context, request datacatalog.ListDatasetsRequest) (*datacatalog.ListDatasetsResponse, error) {
+	err := validators.ValidateListDatasetsRequest(&request)
+	if err != nil {
+		logger.Warningf(ctx, "Invalid list datasets request %v, err: %v", request, err)
+		dm.systemMetrics.validationErrorCounter.Inc(ctx)
+		return nil, err
+	}
+
+	// Get the list inputs
+	listInput, err := transformers.FilterToListInput(ctx, common.Dataset, request.GetFilter())
+	if err != nil {
+		logger.Warningf(ctx, "Invalid list datasets request %v, err: %v", request, err)
+		dm.systemMetrics.validationErrorCounter.Inc(ctx)
```

Comment:
is this the right counter to increment?

Reply:
Yea because when transforming the filters for the dataset, we do some validation too.